### PR TITLE
Feature 21072 catalogue custom fields

### PR DESCRIPTION
**Pull request by Yurujai**

### CODE REVIEW

File: src/Pumukit/NewAdminBundle/Controller/UNESCOController.php [hunk 1]
```diff
@@ -749,6 +619,54 @@ public function cloneAction($multimediaObjectId)
...
+     * @return JsonResponse
+     */
+    public function setCustomFields(Request $request)
+    {
+        $customFields = array_filter($request->request->all(), function ($value) {
+            return -1 != $value;
+        });
+
+        if (!$customFields) {
+            return new JsonResponse(array('success'));
```
Comment (rubenrua) [inline src/Pumukit/NewAdminBundle/Controller/UNESCOController.php]: success?

File: src/Pumukit/SchemaBundle/Document/MultimediaObject.php [hunk 2]
```diff
@@ -417,6 +417,25 @@ public function getType()
         return $this->type;
     }
 
+    /**
+     * @param $type
+     *
+     * @return string
+     */
+    public function getStringType($type)
```
Comment (rubenrua) [inline src/Pumukit/SchemaBundle/Document/MultimediaObject.php]: Check `Job::$statusTexts`

File: src/Pumukit/SchemaBundle/Document/MultimediaObject.php [hunk 3]
```diff
@@ -436,6 +455,26 @@ public function getStatus()
     {
         return $this->status;
     }
+    /**
+     * @param $status
+     *
+     * @return string
+     */
+    public function getStringStatus($status)
```
Comment (rubenrua) [inline src/Pumukit/SchemaBundle/Document/MultimediaObject.php]: Check Job::$statusTexts

File: src/Pumukit/NewAdminBundle/Services/TagCatalogueService.php [hunk 4]
```diff
@@ -0,0 +1,677 @@
...
+    public function addSessionCriteria(Request $request, SessionInterface $session)
+    {
+        $formBasic = false;
+        $newCriteria = [];
+        $tag = [];
+
+        $criteria = $request->request->get('criteria');
+
+        $session->set('admin/unesco/text', false);
+        if ($criteria) {
```
Comment (rubenrua) [inline src/Pumukit/NewAdminBundle/Services/TagCatalogueService.php]: Necessary?

File: src/Pumukit/NewAdminBundle/Services/TagCatalogueService.php [hunk 5]
```diff
@@ -0,0 +1,677 @@
...
+    /**
+     * @return array
+     */
+    public function getAllCustomListFields()
+    {
+        $allFields = [
+            'id' => [
+                'label' => $this->translator->trans('Video ID'),
+                'render' => 'text',
+                'render_params' => array(
```
Comment (rubenrua) [inline src/Pumukit/NewAdminBundle/Services/TagCatalogueService.php]: Is `render_params` necessary?

File: src/Pumukit/NewAdminBundle/Controller/UNESCOController.php (hunk 1 above)
Comment (Yurujai) [reply src/Pumukit/NewAdminBundle/Controller/UNESCOController.php]: Yes, if there is no modification, the default values ​​are maintained

File: src/Pumukit/NewAdminBundle/Services/TagCatalogueService.php (hunk 5 above)
Comment (Yurujai) [reply src/Pumukit/NewAdminBundle/Services/TagCatalogueService.php]: Yes, render_params have sort param to select sortable fields

**Pull request merged**